network-area-diagram/src/main/java/com/powsybl/nad/svg/iidm/DefaultLabelProvider.java

```diff
@@ -94,7 +93,14 @@ public String getBusDescription(BusNode busNode) {
 
     @Override
     public List<String> getVoltageLevelDetails(VoltageLevelNode vlNode) {
-        return Collections.emptyList();
+        List<String> voltageLevelDetails = new ArrayList<>();
+        long activeProduction = Math.round(network.getVoltageLevel(vlNode.getEquipmentId()).getGeneratorStream().mapToDouble(generator -> !Double.isNaN(-generator.getTerminal().getP()) ? -generator.getTerminal().getP() : 0).sum());
+        long reactiveProduction = Math.round(network.getVoltageLevel(vlNode.getEquipmentId()).getGeneratorStream().mapToDouble(generator -> !Double.isNaN(-generator.getTerminal().getQ()) ? -generator.getTerminal().getQ() : 0).sum());
```

Comment:
`network.getVoltageLevel(vlNode.getEquipmentId())` is called four times for each voltage level, you can factorize it